File: LanguAI.Frontend/src/api/models/post-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { AccessEnum } from '../models/access-enum';
import { CommentViewModel } from '../models/comment-view-model';
import { ImageViewModel } from '../models/image-view-model';
export interface PostViewModel {
  access?: AccessEnum;
  comments?: Array<CommentViewModel> | null;
  content?: string | null;
  created?: string;
  id?: number | null;
  image?: ImageViewModel;
  liked?: boolean;
  numberOfComments?: number;
  numberOfLikes?: number;
  username?: string | null;
}

File: LanguAI.Backend/LanguAI.Backend/ViewModels/Post/PostViewModel.cs
using LanguAI.Backend.Core.Enums;
using LanguAI.Backend.ViewModels.Interaction;

namespace LanguAI.Backend.ViewModels.Post;

public class PostViewModel
{
    public int? Id { get; set; }

    public string Content { get; set; }

    public string Username { get; set; }

    public DateTime Created { get; set; }

    public AccessEnum Access { get; set; }

    public bool Liked { get; set; } = false;

    public int NumberOfLikes { get; set; }

    public int NumberOfComments { get; set; }

    public List<CommentViewModel> Comments { get; set; }
}
